server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java

```diff
@@ -86,15 +91,26 @@ public RestSendToExtensionAction(
 
         List<Route> restActionsAsRoutes = new ArrayList<>();
         for (String restAction : restActionsRequest.getRestActions()) {
-            int delim = restAction.indexOf(' ');
+            Optional<String> name = Optional.empty();
+            String[] parts = restAction.split(" ");
+            if (parts.length < 2) {
+                throw new IllegalArgumentException("REST action must contain at least a REST method and route");
+            }
             try {
-                method = RestRequest.Method.valueOf(restAction.substring(0, delim));
-                path = pathPrefix + restAction.substring(delim).trim();
+                method = RestRequest.Method.valueOf(parts[0].trim());
+                path = pathPrefix + parts[1].trim();
+                if (parts.length > 2) {
+                    name = Optional.of(parts[2].trim());
+                }
             } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
                 throw new IllegalArgumentException(restAction + " does not begin with a valid REST method");
             }
             logger.info("Registering: " + method + " " + path);
-            restActionsAsRoutes.add(new Route(method, path));
+            if (name.isPresent()) {
+                restActionsAsRoutes.add(new ProtectedRoute(method, path, name.get()));
```

Comment:
We need to make sure the name is indeed unique here.

Reply:
Yes, we definitely need to ensure there are no collisions with other routes or with transport actions from OpenSearch and plugins. I'm looking at this now and the change looks to be non-trivial. Do you think this should be addressed in this PR, or could it be addressed is a follow-up PR and file an issue to track the work?

Reply:
> Do you think this should be addressed in this PR, or could it be addressed is a follow-up PR and file an issue to track the work?

Leaving the decision to you, personal preference - would be great to see it included into this pull request (pretty coherent change with respect to the context), but if that is non-trivial - please feel free to tackle it separately. Thank you.

Reply:
Hey @owaiskazi19 and @reta I have been working on addressing this comment in another branch and opened up a PR on my local fork to show the changes that it may require: https://github.com/cwperks/OpenSearch/pull/65

Do you think I should merge those changes into this PR and include the check to ensure that the registered name is unique across all registered actions?

Reply:
> Do you think I should merge those changes into this PR and include the check to ensure that the registered name is unique across all registered actions?

I think we could merge those and have a follow up pull request to check for uniqueness (just had a minor comment for changelog entry), thank you

Reply:
@reta with the new protection rule. Need to resolve the comment if it's LGTY.